sqlmesh/core/renderer.py

```diff
@@ -235,13 +234,19 @@ def _expand(node: exp.Expression) -> exp.Expression:
 
         # Ensure there is no data leakage in incremental mode by filtering out all
         # events that have data outside the time window of interest.
-        if add_incremental_filter:
-            # expansion copies the query for us. if it doesn't occur, make sure to copy.
-            if not expand:
-                query = query.copy()
-            for node, _, _ in query.walk(prune=lambda n, *_: isinstance(n, exp.Select)):
-                if isinstance(node, exp.Select):
-                    self.filter_time_column(node, *dates[0:2])
+        if add_incremental_filter and self._time_column:
+            "Filters a query on the time column to ensure no data leakage when running in incremental mode."
+            # expand does a copy already
+            query = t.cast(exp.Subqueryable, query if expand else query.copy())
+            with_ = query.args.pop("with", None)
+            query = (
+                exp.select(*(["*"] if query.is_star else query.named_selects))
```

Comment:
Why can't it always be `*`?

Reply:
my first attempt did this pre column_types and so i needed this expanded, but i've removed that so i can move back to star

Reply:
I think it'd just remove unnecessary details from the code and will make it easier to read